packages/jsii-pacmak/lib/targets/golang/types/class.ts

```diff
@@ -56,62 +105,22 @@ export class GoClass extends GoStruct {
     }
   }
 
-  private emitConstructor(code: CodeMaker): void {
-    const initializer = this.type.initializer;
-    if (initializer) {
-      const constr = `New${this.name}`;
-
-      let params: string[] = [];
-      if (initializer.parameters) {
-        params = initializer.parameters.map(
-          // TODO: resolve types for parameters
-          (x) => `${x.name} ${x.type.toString()}`,
-        );
-      }
-      const parameters = params.length === 0 ? '' : params.join(', ');
-
-      let docstring = '';
-      if (initializer.docs.summary) {
-        docstring = initializer.docs.toString();
-        code.line(`// ${docstring}`);
-      }
-
-      code.openBlock(`func ${constr}(${parameters}) ${this.interfaceName}`);
-
-      // FIXME placeholder
-      code.open(`jsii.NoOpRequest(jsii.NoOpApiRequest {`);
-      code.line(`Class: "${this.name}",`);
-      code.line(`Method: "${constr}",`);
-      code.line(`Parameters: []string{${parameters}}`);
-      code.close(`})`);
-
-      code.line();
-
-      // Return value
-      code.line(`return &${this.name}{`);
-      code.line(` // props`);
-      code.line(`}`);
-      code.closeBlock();
-      code.line();
-    }
+  public get dependencies(): Package[] {
+    return [...super.dependencies, ...getFieldDependencies(this.methods)];
   }
 }
 
 export class ClassMethod implements TypeField {
```

Comment:
Just to verify -- what was the utility of the `TypeField` interface again?

Reply:
TypeField serves as a common interface for resolving module dependencies of referenced types in class/interface properties.